### PR TITLE
Add helper scripts (install nightly ART operator bundles & capture NM logs)

### PR DESCRIPTION
**Pull request by creydr**
**Is this a BUG FIX or a FEATURE ?**:

/kind enhancement

**What this PR does / why we need it**:
This PR adds 2 helper scripts which I used often when working with k-nmstate on OCP:

* `hack/ocp-devscripts-nm-logging.sh`: Script which captures the logs of network manager, network manager dispatcher and the entire journal logs in the background and saves them to disk. This can be helpful if some issues with NM arise and logs needs to attached to a case. In its current state, the script can only be run on the dev-script host machine.
* `hack/ocp-install-nightly-art-operators.sh`: This is an automation of https://hackmd.io/0vbbJdZ7RXem69vCWI61ug to install the latest nightly ART build of the operator bundle (in case you want to test the ART bundles in contrast to using `make ocp-build-and-deploy-bundle` (from 43df3463cfca8febc398e680a07e5942b325a30a)).

**Special notes for your reviewer**:
The scripts don't cover all corner cases (e.g. `hack/ocp-install-nightly-art-operators.sh` if the user has multiple tokens available and the required one is not on 1st place in the array), but the scripts helped me often and I thought it might be worth sharing them.

**Release note**:
```release-note
none
```

### CODE REVIEW

File: hack/ocp-install-nightly-art-operators.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    if [[ $(oc get no | grep SchedulingDisabled) ]]; then
+      # reset timer
+      i=20
+      echo "found node with SchedulingDisabled. Resetting timer..."
+    else
+      echo "Waiting ${i} times to ensure all nodes are ready..."
+      i=$((i - 1))
+    fi
+
+    sleep 1
```
Comment (dougsland) [inline hack/ocp-install-nightly-art-operators.sh:23]: should be sleep ${i} ?

File: hack/ocp-install-nightly-art-operators.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  done
+}	
+
+brew_credentials=$(curl --negotiate -u : https://employee-token-manager.registry.redhat.com/v1/tokens -s) #make sure you created a token before (https://source.redhat.com/groups/public/teamnado/wiki/brew_registry)
```
Comment (dougsland) [inline hack/ocp-install-nightly-art-operators.sh:27]: how about to test if command fails and show the message:

"#make sure you created a token before (https://source.redhat.com/groups/public/teamnado/wiki/brew_registry)" 
exit 1

File: hack/ocp-install-nightly-art-operators.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+brew_username=$(echo $brew_credentials | jq -r ".[0].credentials.username")
+brew_password=$(echo $brew_credentials | jq -r ".[0].credentials.password")
+
+cluster_version=$(oc get clusterversion version -o jsonpath='{.status.desired.version}' | cut -d '.' -f 1,2)
```
Comment (dougsland) [inline hack/ocp-install-nightly-art-operators.sh:31]: for all command being used inside "$( )" I would check if it fails or not. If fails, shows a message and exit 1

**Review by dougsland — COMMENTED** [review]
I am aware this is `hack` dir but added few comments.

File: hack/ocp-install-nightly-art-operators.sh (hunk 3 above)
Comment (cybertron) [reply hack/ocp-install-nightly-art-operators.sh:31]: This script is set -e. If anything fails it will exit immediately.

We might want to add -o pipefail to make sure if a command in a pipeline fails we will catch that too, but we don't need an explicit return value check for every command.

File: hack/ocp-install-nightly-art-operators.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+oc patch OperatorHub cluster --type json -p '[{"op": "add", "path": "/spec/disableAllDefaultSources","value": true}]'
+oc get secret/pull-secret -n openshift-config -o json | jq -r '.data.".dockerconfigjson"' | base64 -d > authfile
+podman login --authfile authfile --username "${brew_username}" --password "${brew_password}" brew.registry.redhat.io
```
[the remark this answers is not in the document]
Comment (cybertron) [reply hack/ocp-install-nightly-art-operators.sh:35]: I guess, although I will note that authfile doesn't _need_ to be removed. The original instructions didn't.

File: hack/ocp-install-nightly-art-operators.sh (hunk 1 above)
Comment (creydr) [reply hack/ocp-install-nightly-art-operators.sh:23]: I want to sleep each iteration one second. `i` is just the counter for the iterations.

File: hack/ocp-install-nightly-art-operators.sh (hunk 3 above)
Comment (creydr) [reply hack/ocp-install-nightly-art-operators.sh:31]: As Ben said, I use `set -e` at the beginning. Anyhow adding the pipefail too, makes sense.

File: hack/ocp-install-nightly-art-operators.sh [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  done
+}	
+
+brew_credentials=$(curl --negotiate -u : https://employee-token-manager.registry.redhat.com/v1/tokens -s) #make sure you created a token before (https://source.redhat.com/groups/public/teamnado/wiki/brew_registry)
+brew_username=$(echo $brew_credentials | jq -r ".[0].credentials.username")
+brew_password=$(echo $brew_credentials | jq -r ".[0].credentials.password")
+
+cluster_version=$(oc get clusterversion version -o jsonpath='{.status.desired.version}' | cut -d '.' -f 1,2)
+
+oc patch OperatorHub cluster --type json -p '[{"op": "add", "path": "/spec/disableAllDefaultSources","value": true}]'
```
[the remark this answers is not in the document]
Comment (creydr) [reply hack/ocp-install-nightly-art-operators.sh:33]: same as above with `set -e`

File: hack/ocp-install-nightly-art-operators.sh [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+oc get secret/pull-secret -n openshift-config -o json | jq -r '.data.".dockerconfigjson"' | base64 -d > authfile
+podman login --authfile authfile --username "${brew_username}" --password "${brew_password}" brew.registry.redhat.io
+oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=authfile
+rm authfile
+oc patch image.config.openshift.io/cluster --type merge -p '{"spec":{"registrySources":{"insecureRegistries":["registry-proxy.engineering.redhat.com"]}}}'
```
[the remark this answers is not in the document]
Comment (creydr) [reply hack/ocp-install-nightly-art-operators.sh:38]: same as above with `set -e`

File: hack/ocp-install-nightly-art-operators.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (creydr) [reply hack/ocp-install-nightly-art-operators.sh]: :+1 addressed it

File: hack/ocp-install-nightly-art-operators.sh [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+brew_credentials=$(curl --negotiate -u : https://employee-token-manager.registry.redhat.com/v1/tokens -s) #make sure you created a token before (https://source.redhat.com/groups/public/teamnado/wiki/brew_registry)
+brew_username=$(echo $brew_credentials | jq -r ".[0].credentials.username")
+brew_password=$(echo $brew_credentials | jq -r ".[0].credentials.password")
+
+cluster_version=$(oc get clusterversion version -o jsonpath='{.status.desired.version}' | cut -d '.' -f 1,2)
+
+oc patch OperatorHub cluster --type json -p '[{"op": "add", "path": "/spec/disableAllDefaultSources","value": true}]'
+oc get secret/pull-secret -n openshift-config -o json | jq -r '.data.".dockerconfigjson"' | base64 -d > authfile
+podman login --authfile authfile --username "${brew_username}" --password "${brew_password}" brew.registry.redhat.io
+oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=authfile
```
[the remark this answers is not in the document]
Comment (creydr) [reply hack/ocp-install-nightly-art-operators.sh]: :+1 addressed it

**Review by dougsland — COMMENTED** [review]
/lgtm
